bin/agent-data-plane/src/main.rs

```diff
@@ -106,17 +109,21 @@ async fn run(started: Instant, logging_api_handler: LoggingAPIHandler) -> Result
 
     // Create a simple pipeline that runs a DogStatsD source, an aggregation transform to bucket into 10 second windows,
     // and a Datadog Metrics destination that forwards aggregated buckets to the Datadog Platform.
-    let blueprint = create_topology(&configuration, env_provider, &component_registry)?;
+    let blueprint = create_topology(&configuration, env_provider, &component_registry).await?;
 
     // Build our administrative API server.
     let primary_api_listen_address = configuration
         .try_get_typed("api_listen_address")
         .error_context("Failed to get API listen address.")?
         .unwrap_or_else(|| ListenAddress::Tcp(([0, 0, 0, 0], 5100).into()));
 
+    let remote_agent_server = new_remote_agent_server()?;
```

Comment:
I don't love this, but it also doesn't matter _right now_ because it's basically a no-op anyways.

This will matter more, though, when we actually need to wire up the gRPC server implementation to the data collected by the component, at which point we'll need to do something more like grab this from the component configuration instead, so that we can build it pre-wired to the component.

Again, not a concern for _this_ PR, but we'll have to do something else for sure when actually adding the logic.